bindings/rust/extended/s2n-tls/src/psk.rs

```diff
@@ -0,0 +1,253 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::ptr::NonNull;
+
+use crate::{
+    enums::PskHmac,
+    error::{Error, ErrorType, Fallible},
+};
+use s2n_tls_sys::*;
+
+#[derive(Debug)]
+pub struct Builder {
+    psk: Psk,
+    has_identity: bool,
+    has_secret: bool,
+    has_hmac: bool,
+}
+
+impl Builder {
+    pub fn new() -> Result<Self, crate::error::Error> {
+        crate::init::init();
+        let psk = Psk::allocate()?;
+        Ok(Self {
+            psk,
+            has_identity: false,
+            has_secret: false,
+            has_hmac: false,
+        })
+    }
+
+    /// Set the public PSK identity.
+    ///
+    /// Corresponds to [s2n_psk_set_identity].
+    pub fn with_identity(&mut self, identity: &[u8]) -> Result<&mut Self, crate::error::Error> {
```

Comment:
One last naming nit: I think our other builders use "set" instead of "with". Any reason this Builder should be different?

Reply:
Usually `with_*` takes ownership of `self` and returns `Self`.